### ISSUE TITLE
Push metrics to cloudwatch Errors when graceful shut down

### ISSUE THREAD

**Issue by xzhouhse** [awslabs/amazon-kinesis-client#1023]
Hi. We have error logs that keep showing up when service instance shut down, error message is:

`Unknown exception while publishing 4 datums to CloudWatch`, stack trace show some more details for the error:
`Unable to execute HTTP request: executor not accepting a task`

We are using graceful shut down in the Kinesis config, when  @PreDestroy hook is triggered,
we call startGracefulShutdown method for each stream consumers

I checked the source code and my expectation is that: CloudWatchMetricsPublisher is still trying to publish to cloudwatch before the metricsFactory shut down. But the request failed because "executor not accepting a task"  anymore after graceful shut down is triggered. Do u already know about this issue or do u have a way to resolve it?  Thx a lot for your help.

What I have tried: update the KCL version(not working), try to handle such exception(not possible because it is from another thread cw-metric-publisher), try to tuning the config for metricsFactory (have not tried yet)

**Comment by jakub-rogowski** [comment]
Hi @xzhouhse have you resolved the issue?

**Comment by xzhouhse** [comment]
@jakub-rogowski no, but maybe u can try the[ latest release of KCL Release 2.6.0](https://github.com/awslabs/amazon-kinesis-client/releases/tag/v2.6.0) which include the [fix for graceful shut down](https://github.com/awslabs/amazon-kinesis-client/pull/1302). I have not tried it yet.
